src/darwin/Framework/CHIP/MTROperationalCredentialsDelegate.mm

```diff
@@ -65,15 +65,8 @@
     // Make copies of the certificates, just in case the API consumer
     // has them as MutableData.
     mRootCert = [NSData dataWithData:rootCert];
-    if (mRootCert == nil) {
-        return CHIP_ERROR_NO_MEMORY;
-    }
-
     if (icaCert != nil) {
         mIntermediateCert = [NSData dataWithData:icaCert];
```

Comment:
This could likely become `mIntermediateCert = [icaCert copy];` without even nil-checking icaCert....

Reply:
Yes, I can do those... we probably don't need to worry about the hypothetically overridden `copy` method

Reply:
done